module-db/Interface/ContactRecord.cpp

```diff
@@ -1559,3 +1559,12 @@ auto ContactRecordInterface::verifyTemporary(ContactRecord &record) -> bool
 
     return isTemporary;
 }
+
+auto ContactRecordInterface::matchedNumberRefersToTemporary(const ContactNumberHolder &matchedNumber) -> bool
+{
+    auto contact = GetByNumberID(matchedNumber.getNumberID());
+    if (not contact.has_value()) {
```

Comment:
[minor]
Could be even:
`return contact.has_value() && contact->isTemporary();`